shell/platform/linux/fl_view.cc

```diff
@@ -89,6 +89,15 @@ G_DEFINE_TYPE_WITH_CODE(
                 G_IMPLEMENT_INTERFACE(fl_text_input_view_delegate_get_type(),
                                       fl_view_text_input_delegate_iface_init))
 
+// Signal handler for GtkWidget::delete-event
+static gboolean window_delete_event_cb(GtkWidget* widget,
+                                       GdkEvent* event,
+                                       FlView* self) {
+  fl_platform_plugin_request_app_exit(self->platform_plugin);
+  // Stop the event from propagating.
+  return TRUE;
```

Comment:
This line totally blocks the additional listener provided by plugins, such like `window_manager`. :(

Reply:
Yes, I agree, it does, but the alternative is returning FALSE, which would not give the app a chance to stop the exit when the last window goes away.

Perhaps this should only make the exit request and return TRUE if there is only one window open. That would still block the plugin listeners on the last window, though.

Reply:
I found the change prevents `*_plugin_dispose(GObject* object)` from being called.

The `some_plugin_dispose` in the generated template will never be called.

```
static void some_plugin_dispose(GObject* object) {
  G_OBJECT_CLASS(some_plugin_parent_class)->dispose(object);
}

static void some_plugin_class_init(SomePluginClass* klass) {
  G_OBJECT_CLASS(klass)->dispose = some_plugin_dispose;
}
```